Core/Core/Serialisation/DeserializationWorkerThreads.cs

```diff
@@ -57,6 +57,11 @@ protected override void ThreadMain()
       catch (Exception ex)
       {
         tcs.SetException(ex);
+
+        if (ex.IsFatal())
+        {
+          throw;
```

Comment:
Not sure what happens (in practice) when we throw here. I have a feeling it becomes an unobserved task exception. Not ideal, so I thought it would be useful to make sure the task completion source `tcs` stays in a faulted state

Reply:
I'm going to have to defer this to @BovineOx 😅

Reply:
This one will need some discussion. I suggest we add anew ticket, technical debt and discuss after the Christmas break